core/src/main/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoEntityProvider.scala

```diff
@@ -409,6 +414,36 @@ class DataRepoEntityProvider(snapshotModel: SnapshotModel, requestArguments: Ent
   override def batchUpdateEntities(entityUpdates: Seq[EntityUpdateDefinition]): Future[Traversable[Entity]] =
     throw new UnsupportedEntityOperationException("batch-update entities not supported by this provider.")
 
-  override def batchUpsertEntities(entityUpdates: Seq[EntityUpdateDefinition]): Future[Traversable[Entity]] =
-    throw new UnsupportedEntityOperationException("batch-upsert entities not supported by this provider.")
+  override def batchUpsertEntities(entityUpdates: Seq[EntityUpdateDefinition]): Future[Traversable[Entity]] = {
+    // TODO AS-770: validate incoming EntityUpdateDefinitions (disallow deletes, etc)
+    // TODO AS-770: translate incoming EntityUpdateDefinitions into key-value writes, including destination BQ datatypes
+    // TODO AS-770: determine destination BQDL dataset, based on snapshot reference. Currently in SnapshotService.generateDatasetName
+
+    // create DeltaInsert object
+    // TODO AS-770: populate with everything we need, including model versioning
+    val dest = Destination(workspaceId = requestArguments.workspace.workspaceIdAsUUID,
+                           referenceId = dataReference.getReferenceId)
+    val ins = DeltaInsert(version = "v0",
+                          insertId = UUID.randomUUID(),
+                          insertTimestamp = new DateTime(),
+                          insertingUser = requestArguments.userInfo.userSubjectId,
+                          destination = dest,
+                          inserts = entityUpdates)
+
+    // consider making this async, so we respond to the user quicker. For now, leave as synchronous
+    // so we return any errors
+    deltaLayerWriter.writeFile(ins) map { _ =>
+      Seq.empty[Entity]
+    } recover {
+      case se: StorageException =>
+        val throwCode = StatusCodes.getForKey(se.getCode).getOrElse(StatusCodes.InternalServerError)
+        throw new DeltaLayerException(s"StorageException in Delta Layer: ${se.getMessage}", se, throwCode)
+      case t: Throwable =>
+        throw new DeltaLayerException(s"Error in Delta Layer: ${t.getClass.getName}: ${t.getMessage}", t)
+    }
+    // This method signature claims to return Traversable[Entity] - and we leave it that way for compatibility -
+    // but note that in practice we don't return any entities. That's good - we don't have access to the
+    // updated entities yet because they will be updated asynchronously by the code that reads the file
+    // we just wrote.
```

Comment:
Sigh. This makes me feel like we're shoehorning delta layer functionality into existing entity code - but that's a discussion separate from this PR.